webClient/src/app/core/menu-bar/menu-bar.component.ts

```diff
@@ -158,18 +158,18 @@ export class MenuBarComponent implements OnInit {
   graphicDiagram() {
     let file = this.editorControl.openFileList.getValue().filter(x => x.active === true)[0];
     if (!file) {
-      this.snackBar.open(`Please open a file before you generate a diagram.`, 'Close', { duration: 2000, panelClass: 'center' });
+      this.snackBar.open(`Please open a file before you generate a diagram.`, 'Close', { duration: 10000, panelClass: 'center' });
```

Comment:
Good duration, but as this file is big enough as it is, I'd much prefer these repeated numbers be a constant to refer to elsewhere.
For numbered constants, we use CAPITAL_AND_UNDERSCORES. 
There's a shared folder within webclient, so collecting constants in a single file may be appropriate.